### DISCUSSION TITLE
How to accept/enable notification permissions on iOS emulator when performing an integration test

### DISCUSSION THREAD

**Discussion by brtbrt** [codemagic-ci-cd/codemagic-docs#1690] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Having a flutter project asking notifications 
```dart
  NotificationSettings settings = await messaging.requestPermission(
    alert: true,
    announcement: false,
    badge: true,
    carPlay: false,
    criticalAlert: false,
    provisional: false,
    sound: true,
  );
```
2. Performing an integration e-2-e test which executes that permission request

### Expected results

No popup asking notification and blocking the app is shown. The app will run normally without blocking

### Actual results

The app awaits the user to tap the "allow" button on the virtual device. The screenshot test fails

### Build id (optional)

6468e3afaed875ed523825be

**Comment by icarusdust** [comment]
Hi @brtbrt, it sounds like more about a Flutter platform specific query. Perhaps, you need to enable permissions for notifications or anything really.

**Reply by brtbrt** [reply]
yep, definitively. After some trials I made it work with https://github.com/leancodepl/patrol which allows interacting with the whole device, not only with the app itself. thanks for answering!
